webapp/src/components/common/Form/index.tsx

```diff
@@ -238,14 +238,14 @@ function Form<TFieldValues extends FieldValues, TContext>(
     return () => {
       window.removeEventListener("beforeunload", listener);
     };
-  }, [t, isDirty]);
+  }, [t, isInProgress, isDirty]);
 
   useUpdateEffect(() => onStateChange?.(formState), [formState]);
 
   useEffect(() => setRef(apiRef, formApiPlus));
 
-  usePrompt(t("form.submit.inProgress"), preventClose.current);
-  usePrompt(t("form.changeNotSaved"), isDirty);
+  usePrompt(t("form.submit.inProgress"), isInProgress);
+  usePrompt(t("form.changeNotSaved"), isDirty && !isInProgress);
```

Comment:
optional: maybe you can wrap those in a useEffect, use strict conditions, and add a cleanup for the prompt
(this can potentially lead to overlapping or competing prompts)

```javascript
  useEffect(() => {
    if (isSubmitting) {
      usePrompt(t("form.submit.inProgress"), true);
    } else if (isDirty) {
      usePrompt(t("form.changeNotSaved"), true);
    } else {
      usePrompt(null, false); // Clear the prompt when no conditions are met
    }

    // Cleanup function to clear the prompt when the component unmounts
    return () => usePrompt(null, false);
  }, [isSubmitting, isDirty, t]);
```